airbyte-webapp/src/hooks/services/Analytics/useTrackPageAnalytics.tsx

```diff
@@ -1,20 +1,19 @@
-import React, { useEffect } from "react";
+import { useEffect } from "react";
 
 import useRouter from "hooks/useRouter";
 
 import { getPageName } from "./pageNameUtils";
 import { useAnalyticsService } from "./useAnalyticsService";
 
-export const TrackPageAnalytics: React.FC = () => {
+export const useTrackPageAnalytics = () => {
```

Comment:
This should never have been a component. I did verify and test this. I had to make changes here to fix a weird build error and changed it to see if it would fix it.
Since I felt it was an improvement I left it be.

Reply:
I'd second Edmundo here. This PR is already extremly large as it is. Let's please not clutter it with additional changes like this. I've added this refactoring to https://github.com/airbytehq/airbyte/pull/12339 to fix that issue separately (since we also want to have that fix in quicker, and not necessarily wait for this PR to be ready).